### ISSUE TITLE
Component `Button` does not meet WCAG 1.4.13: Content on Hover or Focus

### ISSUE THREAD

**Issue by liunate** [carbon-design-system/carbon#8205]
## Component `Button` does not meet WCAG 1.4.13: Content on Hover or Focus

`Button` does not properly implement the [WCAG 1.4.13 Content on Hover or Focus](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) and  [accessibility requirement](https://www.ibm.com/able/requirements/requirements/#1_4_13). 

It is expected keystroke `ESC` should dismiss additional content when triggering element is focused/hovered; otherwise other page content may be covered by the additional content and this is especially a concern for limited vision user with magnifier where their visual readability is severely reduced and hard to pan virtual viewport with the additional content takes big portion of the screen.

I verified latest [carbon storybook Button on "+" button](https://react.carbondesignsystem.com/?path=/story/components-button--playground) can see the same.

https://user-images.githubusercontent.com/38996291/112796370-78263580-909c-11eb-9445-4427d17e1e86.mp4

## Environment

> Operating system  

macOS BigSur

> Browser

Chrome latest

> Assistive technology used to verify

`Cmd` + `+` or pinch with two fingers to zoom in
You can notice the limited page content is even more limited by un-dismissable additional content.

## Detailed description

> What version of the Carbon Design System are you using?

Latest carbon storybook website on carbon design system.

> What did you expect to happen?

`ESC` should dismiss the additional content whether the additional content is brought up by either keyboard focus or pointer hover.

> What happened instead?

`ESC` does not dismiss additional content.

> What WCAG 2.1 checkpoint does the issue violate?

 [WCAG 1.4.13 Content on Hover or Focus](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) and  [accessibility requirement](https://www.ibm.com/able/requirements/requirements/#1_4_13)

## Steps to reproduce the issue

- Keyboard focus
1. [carbon storybook Button on "+" button](https://react.carbondesignsystem.com/?path=/story/components-button--playground) 
2. `Tab` thru until focus on `"+"` button and additional content shows up
3. Press `ESC`, it is not dismissed

- Pointer hover
4. `Tab` away and use pointer to hover on `"+"`
5. Press `ESC`, it is not dismissed

**Comment by emyarod** [comment]
this should be resolved in #8022

**Comment by liunate** [comment]
Thank you! I verified netlify storybook in #8022 and it works great 👍
